packages/@aws-cdk-testing/framework-integ/test/aws-cloudfront-origins/test/integ.function-url-origin.ts

```diff
@@ -18,8 +18,25 @@ const fnUrl = fn.addFunctionUrl({
   authType: lambda.FunctionUrlAuthType.NONE,
 });
 
+const oac = new cloudfront.CfnOriginAccessControl(
+  stack,
+  'OriginAccessControl',
+  {
+    originAccessControlConfig: {
+      name: 'sample',
+      originAccessControlOriginType: 'lambda',
+      signingBehavior: 'always',
+      signingProtocol: 'sigv4',
+    },
+  },
+);
+
 new cloudfront.Distribution(stack, 'Distribution', {
-  defaultBehavior: { origin: new origins.FunctionUrlOrigin(fnUrl) },
+  defaultBehavior: {
+    origin: new origins.FunctionUrlOrigin(fnUrl, {
+      originAccessControlId: oac.attrId,
```

Comment:
I think we can add a new unit test instead of updating this integration test. For the unit test I'd suggest:
-  adding a new function to `test-origin.ts` [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-cloudfront/test/test-origin.ts#L29-L33) which returns a test origin with `originAccessControlId` defined
- adding a unit test to `distribution.test.ts` which sets up a distribution using the test origin from above and check that the `Origin` property in `DistributionConfig` contains `OriginAccessControlId` (can reference existing unit test [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-cloudfront/test/distribution.test.ts#L1246-L1272))

Let me know if you need more guidance on this.

Reply:
Hi @gracelu0, thanks for the feedback!

I have removed the integration tests (which your automated CI forced me to write) and I hope I've added the unit tests correctly. It has run for me so let's see what CI will tell us :-)

Thanks for help!